server/py/framework/utils/notifications/notification_pusher.py

```diff
@@ -49,18 +49,47 @@ def resolve_notifications_default_params():
             notification_module.NotificationTypes.git: {},
             notification_module.NotificationTypes.ipython: {},
             notification_module.NotificationTypes.slack: {},
-            notification_module.NotificationTypes.mail: RunNotificationPusher._get_mail_notification_default_params(),
+            notification_module.NotificationTypes.mail: RunNotificationPusher.get_mail_notification_default_params(),
             notification_module.NotificationTypes.webhook: {},
         }
 
     @staticmethod
-    def _get_mail_notification_default_params():
-        if RunNotificationPusher.mail_notification_default_params is not None:
+    def store_mail_notifications_default_params_to_secret(default_params: dict):
+        smtp_config_secret_name = mlrun.mlconf.notifications.smtp.config_secret_name
+        if framework.utils.singletons.k8s.get_k8s_helper().is_running_inside_kubernetes_cluster():
+            try:
+                return framework.utils.singletons.k8s.get_k8s_helper().store_secrets(
+                    smtp_config_secret_name, secrets=default_params
+                )
+            except ApiException as exc:
+                logger.warning(
+                    "Failed to store SMTP configuration secret",
+                    secret_name=smtp_config_secret_name,
+                    body=mlrun.errors.err_to_str(exc.body),
+                )
```

Comment:
move to a designated class that handles the smtp configuration upon provider (k8s secret)